### PR TITLE
feat: include nodePath as argument to template functions in TreeView nodes 

### PR DESCRIPTION
**Pull request by joaoassisb**

### CODE REVIEW

File: src/treeview/TreeView.jsx [hunk 1]
```diff
@@ -39,7 +32,7 @@ function TreeNodes({ childrenPath, depth, draggable, nodes, parentNode, template
           childrenPath={childrenPath}
           depth={depth}
           draggable={draggable}
-          relativePath={`${relativePath}[${index}]`}
+          relativePath={relativePath ? `${relativePath}[${index}]` : `[${index}]`}
```
Comment (joaoassisb) [inline src/treeview/TreeView.jsx:35]: There was a misunderstanding in the last PR, as the nodes are already an array the correct path to the elements in the first level of recursion should be only the index.


Before change:
image

After change: 
image

Comment (joaoassisb) [reply src/treeview/TreeView.jsx:35]: image

**Comment by assisrafael** [comment]
@joaoassisb after analysing this PR I decided to revert your previous commit. I think you should work more in the proposal and describe what use cases this feature will enable.

File: demo/TreeViewExamples.jsx [hunk 2]
```diff
@@ -51,15 +51,36 @@ export function TreeViewExamples() {
...
+
+      <div className="mt-4">
+        <h1 className="h4">Tree View with Form</h1>
+        <Form initialValues={{ nodes }} onSubmit={(newValues) => console.log(newValues)}>
+          <TreeView
+            nodes={nodes}
+            template={(item, index, parentItem, path) => (
+              <em>
+                <p>Path: path</p>
+                <FormInput name={`nodes${path}.attrA`} />
```
Comment (joaoassisb) [inline demo/TreeViewExamples.jsx:76]: The point with this change is to allow the usage of a TreeView with Form Inputs. This behavior is not possible using only the 3 args (item, index, parentItem).

image

**Comment by joaoassisb** [comment]
> @joaoassisb after analysing this PR I decided to revert your previous commit. I think you should work more in the proposal and describe what use cases this feature will enable.
@assisrafael the use case is to simplify the usage of TreeViews with Forms. At the moment, the template function passes only the item,index and parentItem, making a lot harder to handle paths and possible changes.

I updated the TreeViewExample to illustrate better my use case.

image

**Comment by assisrafael** [comment]
> > @joaoassisb after analysing this PR I decided to revert your previous commit. I think you should work more in the proposal and describe what use cases this feature will enable.
> 
> @assisrafael the use case is to simplify the usage of TreeViews with Forms. At the moment, the template function passes only the item,index and parentItem, making a lot harder to handle paths and possible changes.
> 
> I updated the TreeViewExample to illustrate better my use case.
> image
> Now I can use a FormInput using the relativePath in the name of the input, and the changes will automatically updates the initial nodes structure.
> 
> image

You use case makes sense. But I think it would be better with the prop would be called `nodePath` instead of `relativePath`.

File: src/treeview/TreeView.jsx [hunk 3]
```diff
@@ -32,6 +32,7 @@ function TreeNodes({ childrenPath, depth, draggable, nodes, parentNode, template
           childrenPath={childrenPath}
           depth={depth}
           draggable={draggable}
+          relativePath={relativePath ? `${relativePath}[${index}]` : `[${index}]`}
```
Comment (assisrafael) [inline src/treeview/TreeView.jsx:35]: Instead of using a ternary here you could set `""` as the default value for the prop

**Review by assisrafael — APPROVED** [review]

**Pull request merged**